src/cockpit/packages.py

```diff
@@ -57,31 +57,24 @@ def directory_items(path):
 # - rewrite cockpit-askpass in Python and write it into a temporary file
 # - bundle helper shell scripts into their webpacks
 # Until then, we need this libexecdir detection hack.
-LIBEXECDIR = None
-
-
-def get_libexecdir() -> str:
-    '''Detect libexecdir on current machine
-
-    This only works for systems which have cockpit-ws installed.
-    '''
-    global LIBEXECDIR
-    if LIBEXECDIR is None:
-        for candidate in ['/usr/local/libexec', '/usr/libexec', '/usr/local/lib/cockpit', '/usr/lib/cockpit']:
-            if os.path.exists(os.path.join(candidate, 'cockpit-askpass')):
-                LIBEXECDIR = candidate
-                break
-        else:
-            logger.warning('Could not detect libexecdir')
-            # give readable error messages
-            LIBEXECDIR = '/nonexistent/libexec'
-
-    return LIBEXECDIR
+LIBEXEC_PATH = [
+    '/usr/local/libexec',
+    '/usr/libexec',
+    '/usr/local/lib/cockpit',
+    '/usr/lib/cockpit',
+    *os.environ['PATH'].split(':')
+]
 
 
 def patch_libexecdir(obj):
     if isinstance(obj, str):
-        return obj.replace('${libexecdir}', get_libexecdir())
+        before, expansion, after = obj.partition('${libexecdir}/')
+        if expansion:
+            for path in LIBEXEC_PATH:
+                if os.path.exists(f'{path}/{after}'):
+                    return f'{before}{path}/{after}'
+            logger.warning(f'Unable to expand {expansion}{after}')
```

Comment:
This warning now appears in [two tests](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18304-20230207-220051-8dcd1eca-fedora-37-pybridge/log.html#184), somewhat surprisingly -- we don't allow the former "Could not detect libexecdir" message either.

Reply:
The warning is because of `cockpit-ssh` which is apparently not installed in those tests.  Maybe we need a way to evaluate this only at Peer launch time.  That wouldn't be unprecedented.

Reply:
Although I just realize now why we do it at manifest parse time: the "libexecdir" entry in `systemd/manifest.json` which we need to find the certificate helper script...